src/main/java/com/minecolonies/coremod/client/gui/WindowHutAllInventory.java

```diff
@@ -158,29 +165,25 @@ else if (rack instanceof ChestTileEntity)
                     final ItemStack stack = ((ChestTileEntity) rack).getStackInSlot(slot);
                     if (!ItemStackUtils.isEmpty(stack))
                     {
-                        items.add(stack.copy());
+                        ItemStorage currentStorage = new ItemStorage(stack.copy());
+                        if (storedItems.containsKey(currentStorage))
+                        {
+                            storedItems.put(currentStorage, storedItems.get(currentStorage) + stack.getCount());
+                        }
+                        else
+                        {
+                        	storedItems.put(currentStorage, stack.getCount());
+                        }
                     }
                 }
             }
         }
 
-        final Map<ItemStorage, ItemStorage> storedItems = new HashMap<>();
-        for (final ItemStack currentItem : items)
-        {
-            final ItemStorage currentStorage = new ItemStorage(currentItem, currentItem.getCount(), false);
-
-            if (storedItems.containsKey(currentStorage))
-            {
-                final ItemStorage existing = storedItems.get(currentStorage);
-                existing.setAmount(existing.getAmount() + currentStorage.getAmount());
-            }
-            else
-            {
-                storedItems.put(currentStorage, currentStorage);
-            }
-        }
-
-        final List<ItemStorage> filterItems = new ArrayList<>(storedItems.keySet());
+        final List<ItemStorage> filterItems = new ArrayList<>();
+        storedItems.forEach((storage, amount) -> {
+            storage.setAmount(amount);
+            filterItems.add(storage);
+        });
```

Comment:
Is like this OK, or do i have to find another way to set the amount of the ItemStorage?

Reply:
Should be fine like this.
Did you test this?

Reply:
Yes i did.

Reply:
If you want to suggest any more changes, i will look at it tomorrow, today its late enough :)

Reply:
Looks fine for me now. Does this solve all the issues there were before?

Can you add an additional check to query the rack from the position of the building itself (it's not include din the list of containers, so it needs an additional line to add the stacks from it.

Reply:
"Does this solve all the issues there were before?" idk, i don't know about many issues with this view, i just made this to fix the linked issue.
"Can you add an additional check to query the rack from the position of the building itself" I can do that, but i think lexustec said he is working on that.

Reply:
"Can you add an additional check to query the rack from the position of the building itself"
I have a working version for the warehouse now, i will test it with some other buildings, and than hopefully push it.